_posts/2023-10-25-aws-lambda-linode-migrate.md

```diff
@@ -0,0 +1,527 @@
+---
+title: "How to migrate Python functions from AWS Lambda to OpenFaaS and Kubernetes"
+description: "This guide will show you how to migrate existing AWS Lambda functions to OpenFaaS running on LKE"
+date: 2023-10-25
+categories:
+- openfaas
+- akamai
+- linode
+- lambda
+dark_background: true
+author_staff_member: Han
+hide_header_image: true
+---
+
+In this guide we'll show you how to migrate existing AWS Lambda functions to a Akamai Cloud Computing LKE cluster with OpenFaaS.
+
+> Did you know? Linode was acquired by Akamai, and is now being branded as "Akamai Cloud Computing".
+> The rebranding is still in-progress, so we'll be referring to Linode throughout this article.
+
+This makes functions completely portable and allows running
+- Running at any scale with predictable cost.
+- Remove any limits and run workloads AWS Lambda is not suitable for with the 
+- Get a great developer experience build on familiar docker concepts.
+
+We will work through migrating a real-world extract, transform, and load (ETL) workflow from AWS to Linode with OpenFaaS.
+
+## Deploy OpenFaaS on Linode Kubernetes Engine
+
+See their [getting started guide](https://www.linode.com/docs/products/compute/kubernetes/get-started/) to setup an LKE cluster.
+
+### Install OpenFaaS
+
+Install OpenFaaS on the cluster. Choose either the [Community Edition (CE)](https://docs.openfaas.com/deployment/kubernetes/) which is intended for development or proof of concepts.
+
+For production and commercial use you should deploy [OpenFaaS Standard or OpenFaaS for Enterprises](https://docs.openfaas.com/deployment/pro/).
+
+
+## Lambda functions vs OpenFaaS functions
+
```

Comment:
Lambda has the concept of runtimes, OpenFaaS has the concept of templates, and can run any container that serves HTTP or can be invoked as a CLI.

Below, we compare the python3-http OpenFaaS template to the Lambda "python 3.11" runtime.

So, let's start by looking at